javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java

```diff
@@ -95,7 +95,7 @@ public final void write(ClientModel model, JavaFile javaFile) {
         javaFile.declareImport(imports);
 
         javaFile.javadocComment(settings.getMaximumJavadocCommentWidth(),
-            comment -> comment.description(model.getDescription()));
+            comment -> comment.description(model.getDescription(), isPartialUpdateSupported()));
```

Comment:
As we keep adding more one-offs for partial update support, do we just want to enable it always and for everyone? Or at least in documentation, enable it for everyone. It's an extremely useful feature to allow extending documentation without complicated Swagger/TypeSpec transforms or code customizations.

Reply:
Enable for all javadoc could be fine, but we may need to check a few more situations.

E.g. for a full Javadoc, should we do this
```
    /**
     * <!-- start generated doc -->
     * The read operation.
     * <!-- end generated doc -->
     * 
     * @param filter A sequence of textual characters.
     * @throws IllegalArgumentException thrown if parameters fail the validation.
     * @throws HttpResponseException thrown if the request is rejected by server.
     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
     * @return the response.
     */
```

or this
```
    /**
     * <!-- start generated doc -->
     * The read operation.
     * 
     * @param filter A sequence of textual characters.
     * @throws IllegalArgumentException thrown if parameters fail the validation.
     * @throws HttpResponseException thrown if the request is rejected by server.
     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
     * @return the response.
     * <!-- end generated doc -->
     */
```

Reply:
Further discussion on Javadoc
https://github.com/Azure/autorest.java/issues/2510#issuecomment-1897697164